src/adapters/kafka/index.ts

```diff
@@ -0,0 +1,166 @@
+import fs from 'fs'
+import { Kafka, Consumer } from 'kafkajs'
+import Adapter from '../../lib/adapter.js'
+import GleeMessage from '../../lib/message.js'
+
+
+class KafkaAdapter extends Adapter {
+  private kafka: Kafka
+  private firstConnect: boolean = true
+  name(): string {
+    return 'Kafka adapter'
+  }
+
+  async connect() {
+    const broker = this.AsyncAPIServer.url()
+    this.kafka = new Kafka({
+      clientId: 'glee-app',  // clientID: hardcoded need to change afterwards 
+      brokers: [broker],
+    })
+    const consumer = this.kafka.consumer({ groupId: 'glee-group' })   // groupID: hardcoded need to change afterwards
+    consumer.on('consumer.connect', () => {
+      if (this.firstConnect) {
+        this.firstConnect = false
+        this.emit('connect', {
+          name: this.name(),
+          adapter: this,
+          connection: consumer,
+          channels: this.channelNames
+        })
+      }
+    })
+
+    await consumer.connect()
+    
+    const subscribedChannels = this.getSubscribedChannels()
+    await consumer.subscribe({ topics: subscribedChannels, fromBeginning: true })
```

Comment:
Not for now but let's not forget to make "fromBeginning" configurable.

Reply:
yeaa definitely